lib/common/api/deprecate.js

```diff
@@ -18,101 +34,49 @@ const deprecate = {
       return console.warn(`(electron) ${message}`)
     }
   },
-  renameFunction: function (fn, oldName, newName) {
-    let warned = false
-    return function () {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.warn(oldName, newName)
-      }
-      return fn.apply(this, arguments)
-    }
-  },
-  removeFunction: (oldName) => {
-    if (!process.noDeprecation) {
-      deprecate.log(`The '${oldName}' function has been deprecated and marked for removal.`)
-    }
-  },
-  alias: function (object, oldName, newName) {
-    let warned = false
-    const newFn = function () {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.warn(oldName, newName)
-      }
-      return this[newName].apply(this, arguments)
-    }
-    if (typeof object === 'function') {
-      object.prototype[newName] = newFn
-    } else {
-      object[oldName] = newFn
-    }
-  },
+
   event: (emitter, oldName, newName) => {
-    let warned = false
+    const warn = newName.startsWith('-') /* internal event */
+      ? warnOnce(`${oldName} event`)
+      : warnOnce(`${oldName} event`, `${newName} event`)
     return emitter.on(newName, function (...args) {
-      if (this.listenerCount(oldName) === 0) return
-      if (warned || process.noDeprecation) return
-
-      warned = true
-      if (newName.startsWith('-')) {
-        deprecate.log(`'${oldName}' event has been deprecated.`)
-      } else {
-        deprecate.warn(`'${oldName}' event`, `'${newName}' event`)
+      if (this.listenerCount(oldName) !== 0) {
+        warn()
+        this.emit(oldName, ...args)
       }
-      this.emit(oldName, ...args)
     })
   },
-  removeProperty: (object, deprecated, ignoreMissingProps = false) => {
-    let warned = false
-    let warn = () => {
-      if (!(warned || process.noDeprecation)) {
-        warned = true
-        deprecate.log(`The '${deprecated}' property has been deprecated and marked for removal.`)
-      }
-    }
 
-    if (!(deprecated in object)) {
-      if (ignoreMissingProps) return
-      throw new Error('Cannot deprecate a property on an object which does not have that property')
-    }
+  removeProperty: (o, removedName) => {
+    const warn = warnOnce(removedName)
```

Comment:
why was ignoreMissingProps removed? as is, it looks like we would warn regardless of whether the property exists on the object; would we necessarily want that?

Reply:
My thinking is this reminds the maintainers to remove the deprecation because the property no longer exists.

Is there a reason to keep a `deprecate.removeProperty()` call around on a property that's already gone?